### PR TITLE
Add a feature that enables https 

### PR DESCRIPTION
**Pull request by namnx228**
This PR adds a new feature that allows user to enable TLS on the http server that receives alerts from the prometheus alert manager.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,11 @@ To start the receiver execute the command:
 
     $ ./prometheus-webhook-snmp run
 
+To start with https, pass the certificate and key parameters:
```
Comment (votdev) [inline README.md]:
```suggestion
To start with HTTPS, pass the certificate and key parameters:
```

File: README.md [hunk 2]
```diff
@@ -14,6 +14,11 @@ To start the receiver execute the command:
...
+
+    $ ./prometheus-webhook-snmp --cert "certificate.pem" --key "key.pem" run
+
+Note: Both '--cert' and '--key' are needed to enable https
```
Comment (votdev) [inline README.md]:
```suggestion
Note: Both '--cert' and '--key' are needed to enable HTTPS.
```

File: prometheus-webhook-snmp [hunk 3]
```diff
@@ -40,11 +40,15 @@ pass_context = click.make_pass_decorator(utils.Context, ensure=True)
               help='The trap OID if none is found in the Prometheus alert labels.')
 @click.option('--trap-default-severity',
               help='The trap severity if none is found in the Prometheus alert labels.')
+@click.option('--cert',
+        help='The CA certificate for setting up https. Need to be used with --key. Without this option, http is used.')
```
Comment (votdev) [inline prometheus-webhook-snmp]:
```suggestion
        help='The CA certificate for setting up HTTPS. Need to be used with --key. Without this option, HTTP is used.')
```

File: prometheus-webhook-snmp [hunk 4]
```diff
@@ -40,11 +40,15 @@ pass_context = click.make_pass_decorator(utils.Context, ensure=True)
...
+@click.option('--key',
+        help='The CA private key for setting up https. Need to be used with --cert')
```
Comment (votdev) [inline prometheus-webhook-snmp]:
```suggestion
        help='The CA private key for setting up HTTPS. Need to be used with --cert.')
```

**Review by votdev — CHANGES_REQUESTED** [review]

**Comment by namnx228** [comment]
Hi @votdev, 
Your comments have been addressed. Please take a look again.
Thanks for your review

**Comment by namnx228** [comment]
@votdev Please let me know if you have any other comments. Otherwise, this PR can go in.

**Comment by votdev** [comment]
Sorry, this will take some time. I'll have to fix some pylint issues and testing the feature in a container does not work. I think dependencies to SSL libraries in RPC spec is missing, too. The PR needs some refactoring.

**Comment by votdev** [comment]
Sorry, this will take some time. You'll have to fix some pylint issues and testing the feature in a container does not work.
```
$ podman run -v $(pwd)/prometheus-webhook-snmp:/usr/bin/prometheus-webhook-snmp -v $(pwd)/prometheus-webhook-snmp.conf:/etc/prometheus-webhook-snmp.conf -v $(pwd)/cert.pem:/srv/cert.pem -v $(pwd)/key.pem:/srv/key.pem --env RUN_ARGS="--cert /srv/cert.pem --key /srv/key.pem" -p 9099:9099 prometheus-webhook-snmp:latest
```

I think dependencies to SSL libraries in RPM spec is missing, too. The PR needs some refactoring.

```
************* Module nordix-prometheus-webhook-snmp.prometheus_webhook_snmp.utils
prometheus_webhook_snmp/utils.py:189:0: C0330: Wrong hanging indentation (remove 1 space).
             'server.ssl_module': 'builtin',
            |^ (bad-continuation)
prometheus_webhook_snmp/utils.py:190:0: C0330: Wrong hanging indentation (remove 1 space).
             'server.ssl_certificate': ctx.config['cert'],
            |^ (bad-continuation)
prometheus_webhook_snmp/utils.py:191:0: C0330: Wrong hanging indentation (remove 1 space).
             'server.ssl_private_key': ctx.config['key']
            |^ (bad-continuation)
prometheus_webhook_snmp/utils.py:192:0: C0330: Wrong hanging indentation.
         })
        |^  | (bad-continuation)
prometheus_webhook_snmp/utils.py:195:0: C0303: Trailing whitespace (trailing-whitespace)
```

I also think the `--cert`  and `--key` options should be relocated to the `run`  command because they are only used in this context.

I've did some refactoring, but can not push the changes because of `ERROR: Permission to Nordix/prometheus-webhook-snmp.git denied to votdev.`. Did you have set the correct permissions when creating the PR?

**Comment by namnx228** [comment]
@votdev  The Nordix organization restricts the access to their repositories, so only members of the organization can push commit. The folk this PR comes from is in this organization, that is why you cannot push new commit to it. But you can refactor by making change request, and I will update the PR accordingly. 
Meanwhile, I will test the PR with podman container, and fix as what you suggest above. Thanks for reviewing.

**Comment by votdev** [comment]
If you apply this patch, then the PR should be fine.
```
diff --git a/prometheus_webhook_snmp/utils.py b/prometheus_webhook_snmp/utils.py
--- a/prometheus_webhook_snmp/utils.py	(revision a4315bbe3ff29e5df6276073eff5d5c35c089102)
+++ b/prometheus_webhook_snmp/utils.py	(date 1644939839063)
@@ -185,14 +185,14 @@
     :type ctx: dict
     """
     if ctx.config['cert'] and ctx.config['key']:
-         cherrypy.config.update({
-             'server.ssl_module' : 'builtin',
-             'server.ssl_certificate' : ctx.config['cert'],
-             'server.ssl_private_key' : ctx.config['key']
-         })
+        cherrypy.config.update({
+            'server.ssl_module': 'builtin',
+            'server.ssl_certificate': ctx.config['cert'],
+            'server.ssl_private_key': ctx.config['key']
+        })
     elif ctx.config['cert'] or ctx.config['key']:
         raise Exception("Both '--cert' and '--key' are needed")
- 
+
     cherrypy.config.update({
         'environment': 'production',
         'server.socket_host': ctx.config['host'],

diff --git a/prometheus-webhook-snmp b/prometheus-webhook-snmp
--- a/prometheus-webhook-snmp	(revision a4315bbe3ff29e5df6276073eff5d5c35c089102)
+++ b/prometheus-webhook-snmp	(date 1644939829439)
@@ -40,15 +40,11 @@
               help='The trap OID if none is found in the Prometheus alert labels.')
 @click.option('--trap-default-severity',
               help='The trap severity if none is found in the Prometheus alert labels.')
-@click.option('--cert',
-        help='The CA certificate for setting up HTTPS. Need to be used with --key. Without this option, HTTP is used.')
-@click.option('--key',
-        help='The CA private key for setting up HTTPS. Need to be used with --cert.')
 @click.version_option(__version__, message="%(version)s")
 @pass_context
 def cli(ctx, debug, snmp_host, snmp_port, snmp_community, snmp_retries,
         snmp_timeout, alert_oid_label, trap_oid_prefix, trap_default_oid,
-        trap_default_severity, cert, key):
+        trap_default_severity):
     ctx.config.load(click.get_current_context().info_name)
     ctx.config['debug'] = True if debug else None
     ctx.config['snmp_host'] = snmp_host
@@ -60,8 +56,6 @@
     ctx.config['trap_default_oid'] = trap_default_oid
     ctx.config['trap_default_severity'] = trap_default_severity
     ctx.config['trap_oid_prefix'] = trap_oid_prefix
-    ctx.config['cert'] = cert
-    ctx.config['key'] = key
 
     if ctx.config['debug']:
         logging.basicConfig(level=logging.DEBUG)
@@ -78,11 +72,18 @@
 @click.option('--metrics',
               is_flag=True,
               help='Provide Prometheus metrics from this receiver.')
+@click.option('--cert',
+              help='The CA certificate for setting up HTTPS. '
+                   'Need to be used with --key. Without this option, HTTP is used.')
+@click.option('--key',
+              help='The CA private key for setting up HTTPS. Need to be used with --cert.')
 @pass_context
-def run(ctx, host, port, metrics):
+def run(ctx, host, port, metrics, cert, key):
     ctx.config['host'] = host
     ctx.config['port'] = port
     ctx.config['metrics'] = True if metrics else None
+    ctx.config['cert'] = cert
+    ctx.config['key'] = key
     ctx.config.dump()
     utils.run_http_server(ctx)
     sys.exit(0)
```

**Comment by votdev** [comment]
Testing the code in a container requires the following command line, but then the PR + patch seems to work then: `podman run -v $(pwd)/prometheus-webhook-snmp:/usr/bin/prometheus-webhook-snmp -v $(pwd)/prometheus_webhook_snmp/:/lib/python3.8/site-packages/prometheus_webhook_snmp/ -v $(pwd)/prometheus-webhook-snmp.conf:/etc/prometheus-webhook-snmp.conf -v $(pwd)/cert.pem:/srv/cert.pem -v $(pwd)/key.pem:/srv/key.pem --env RUN_ARGS="--cert /srv/cert.pem --key /srv/key.pem" -p 9099:9099 prometheus-webhook-snmp:latest`

**Comment by votdev** [comment]
You can now use `tox -e lint` to check the code and `tox -e fix` to auto-fix it.

**Comment by votdev** [comment]
Looks good now. Thank you.

**Comment by namnx228** [comment]
Thanks for reviewing and merging this PR.

**Pull request merged**